autoendpoint/src/extractors/authorization_check.rs

```diff
@@ -7,14 +7,45 @@ use actix_web::web::Data;
 use actix_web::{FromRequest, HttpRequest};
 use futures::future::LocalBoxFuture;
 use futures::FutureExt;
+use openssl::error::ErrorStack;
 use uuid::Uuid;
 
 /// Verifies the request authorization via the authorization header.
 ///
 /// The expected token is the HMAC-SHA256 hash of the UAID, signed with one of
 /// the available keys (allows for key rotation).
+/// NOTE: This is *ONLY* for internal calls that require authorization and should
+///      NOT be used by calls that are using VAPID authentication (e.g.
+///      subscription provider endpoints)
```

Comment:
what do we mean by "local auth" exactly?

All the registration endpoints use this auth check.

Reply:
There are two types of authentication: VAPID, which is meant for external, 3rd party subscription providers, and "local" User Agent authorization, which uses a bearer token provided as part of the registration process. 

The complication here is that both share the "Authentication" header. This call should only be used to check the latter Authentication form.